### PR TITLE
[functionapp] Enable server side build in OryxBuilder

### PR DESCRIPTION
**Pull request by Hazhzeng**
### Overview
1. Allow Oryx Builder to distinguish Linux Consumption Functionapp Build by "SCM_RUN_FROM_PACKAGE" environment variable.
2. Default the optimization flag of Linux Consumption Functionapp Build to UseExpressBuild.
3. Allow Linux Consumption Functionapp Build to **upload the built content** to a customer's storage account. The package will be uploaded from /home/data/SitePackages/<zip_file> to SCM_RUN_FROM_PACKAGE SAS Uri. (refer to **UploadBuiltContent()** function)
4. Allow Linux Consumption Functionapp Build to **reset all Service Fabric Mesh container workers**. (refer to **RemoveAllWorkers()** function)

### CR Request
@ankitkumarr @maiqbal11 @sanchitmehta

### CODE REVIEW

File: Kudu.Core/Deployment/Generator/OryxBuilder.cs [hunk 1]
```diff
@@ -65,7 +80,13 @@ public override Task Build(DeploymentContext context)
                 }
             }
 
-            return Task.CompletedTask;
+            // Detect if package upload is necessary for server side build
+            if (artifactPath != null)
+            {
```
Comment (ankitkumarr) [inline Kudu.Core/Deployment/Generator/OryxBuilder.cs]: Please correct me if I am wrong, but looks like `artifactPath` will never actually be `null`. So, this path will always be hit, whether it's consumption or dedicated. We probably only want to hit this condition if we are in Consumption I think. Should we do a check for `FunctionAppHelper.HasScmRunFromPackage()` here?

File: Kudu.Core/Helpers/PostDeploymentHelper.cs [hunk 2]
```diff
@@ -348,6 +349,47 @@ public static async Task PerformAutoSwap(string requestId, TraceListener tracer)
...
+            string authTokenEncoded = HttpUtility.UrlEncode(authToken);
+            Trace(TraceEventType.Information, "RemoveAllWorkers, websiteAuthEncryptionKey = {0}", websiteAuthEncryptionKey);
+            Trace(TraceEventType.Information, "RemoveAllWorkers, authToken = {0}", authToken);
+            Trace(TraceEventType.Information, "RemoveAllWorkers, authTokenEncoded = {0}", authTokenEncoded);
+
+            // Generate RemoveAllWorker request URI
+            /* TODO: Hazhzeng, uncomment this when "RemoveAllStandardWorker" is out in ANT 83
+             * string baseUrl = $"http://{websiteHostname}/operations/removeworker/{sitename}/allStandard?token={authTokenEncoded}";
+             */
+            string baseUrl = $"http://{websiteHostname}/operations/removeworker/{sitename}/all?token={authTokenEncoded}";
```
Comment (ankitkumarr) [inline Kudu.Core/Helpers/PostDeploymentHelper.cs]: Maybe https?

File: Kudu.Core/Deployment/Generator/OryxBuilder.cs [hunk 3]
```diff
@@ -40,23 +42,36 @@ public override Task Build(DeploymentContext context)
...
+                }
 
                 //
                 // Run express build setups if needed
                 if (args.Flags == BuildOptimizationsFlags.UseExpressBuild)
                 {
                     if (FunctionAppHelper.LooksLikeFunctionApp())
                     {
-                        SetupFunctionAppExpressArtifacts(context);
+                        artifactPath = SetupFunctionAppExpressArtifacts(context);
```
Comment (ankitkumarr) [inline Kudu.Core/Deployment/Generator/OryxBuilder.cs]: `SetupFunctionAppExpressArtifacts` sets `packaname.txt` and `packagepath.txt` as well. We probably don't need that for consumption. Maybe we just do the packaging for Consumption, and not use `BUILD_FLAGS`.

File: Kudu.Core/Helpers/PostDeploymentHelper.cs (hunk 2 above)
Comment (Hazhzeng) [reply Kudu.Core/Helpers/PostDeploymentHelper.cs]: The internal call only accepts http protocol. Tried with https, failed to create SSL channel :(

Comment (ankitkumarr) [reply Kudu.Core/Helpers/PostDeploymentHelper.cs]: Fair

File: Kudu.Services/LinuxConsumptionInstanceAdmin/LinuxConsumptionInstanceAdminController.cs [hunk 4]
```diff
@@ -37,7 +37,7 @@ public LinuxConsumptionInstanceAdminController(ILinuxConsumptionInstanceManager
         /// </summary>
         /// <returns>Expect 200 when current service is up and running</returns>
         [HttpGet]
-        [Authorize(Policy = AuthPolicyNames.AdminAuthLevel)]
+        //[Authorize(Policy = AuthPolicyNames.AdminAuthLevel)]
```
Comment (sanchitmehta) [inline Kudu.Services/LinuxConsumptionInstanceAdmin/LinuxConsumptionInstanceAdminController.cs]: Do we want to disable the auth here?

**Pull request merged**